### DISCUSSION TITLE
Sharing code between test cases

### DISCUSSION THREAD

**Discussion by ShuffleWire** [cyberbotics/webots#5509] [Q&A]
Hello,
I've noticed that a lot of code in share/duplicated among "pen_***" tests cases. Is there a way to cleanly share code/data/types between those ?

**Comment by omichel** [comment]
I am not sure what could be a good solution to this...

**Reply by ShuffleWire** [reply]
could we add dependency to the "main" pen test, that could hold all the wrappers/utilities, and link against this project (using (c)make dependency management) ? Just an idea, and I definitely not able to do it...

**Reply by omichel** [reply]
That appears complicated, especially if you are not able to do it. I would drop it for now.

**Reply by omichel** [reply]
If you are looking for an idea for a fairly easy contribution, maybe you can try to address this one: #5492.
